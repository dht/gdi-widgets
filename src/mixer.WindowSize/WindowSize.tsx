import React from 'react';
import { useWindowSize } from 'react-use';
import { Wrapper, Height, Row, Times, Units, Width } from './WindowSize.style';

export type WindowSizeProps = {};

export function WindowSize(_props: WindowSizeProps) {
    const { width, height } = useWindowSize();

    return (
        <Wrapper
            className='WindowSize-wrapper'
            data-testid='WindowSize-wrapper'
        >
            <Row>
                <Width>{width.toLocaleString()}</Width>
                <Times>x</Times>
                <Height>{height.toLocaleString()}</Height>
                <Units>px</Units>
            </Row>
        </Wrapper>
    );
}

export default WindowSize;
