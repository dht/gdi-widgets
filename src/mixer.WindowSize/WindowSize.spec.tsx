import { WindowSizeDriver } from './WindowSize.driver';
import Chance from 'chance';

const chance = new Chance();

describe('WindowSize', () => {
    let driver: WindowSizeDriver;

    beforeAll(() => {
        driver = new WindowSizeDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const containerClassName = element.get.containerClassName();
        const innerText = element.get.label();

        expect(containerClassName).toContain('WindowSize-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('WindowSize snapshots', () => {
    let driver: WindowSizeDriver;

    beforeAll(() => {
        driver = new WindowSizeDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
