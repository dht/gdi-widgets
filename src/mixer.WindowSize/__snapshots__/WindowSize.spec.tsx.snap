// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowSize snapshots should match snapshot 1`] = `
<div
  className="sc-bczRLJ hEqbSV WindowSize-wrapper"
  data-testid="WindowSize-wrapper"
>
  <div
    className="sc-gsnTZi cOFMKY"
  >
    <div
      className="sc-dkzDqf chgQAY"
    >
      ∞
    </div>
    <div
      className="sc-hKMtZM hJgazB"
    >
      x
    </div>
    <div
      className="sc-eCYdqJ gierlq"
    >
      ∞
    </div>
    <div
      className="sc-jSMfEi kOeYUx"
    >
      px
    </div>
  </div>
</div>
`;
